src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -345,7 +345,7 @@ private function populateAvailableType($id, Definition $definition)
             unset($this->ambiguousServiceTypes[$type]);
         }
 
-        if ($definition->isDeprecated() || !$reflectionClass = $this->container->getReflectionClass($definition->getClass(), false)) {
+        if (preg_match('/^\d+_[^~]++~[._a-zA-Z\d]{7}$/', $id) || $definition->isDeprecated() || !$reflectionClass = $this->container->getReflectionClass($definition->getClass(), false)) {
```

Comment:
Do we have such a regex somewhere else? Would it make sense to store it as a reusable const somewhere?

Reply:
That's internal to me: I added an integration test ensuring we don't rely on this exact detail, but on the actual behavior. Thus, not sure a const is needed.